pkg/piping/sync_async_piping_manager.go

```diff
@@ -0,0 +1,172 @@
+package piping
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/regionless-storage-service/pkg/config"
+	"github.com/regionless-storage-service/pkg/constants"
+	"github.com/regionless-storage-service/pkg/database"
+	"github.com/regionless-storage-service/pkg/index"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+)
+
+type SyncAsyncPiping struct {
+	databaseType constants.StoreType
+}
+
+func NewSyncAsyncPiping(storeType constants.StoreType) *SyncAsyncPiping {
+	return &SyncAsyncPiping{databaseType: storeType}
+}
+
+func (sap *SyncAsyncPiping) Read(ctx context.Context, rev index.Revision) (string, error) {
+	_, rootSpan := otel.Tracer(config.TraceName).Start(ctx, "SyncAsyncPiping Read")
+	defer rootSpan.End()
+	syncNodes, asyncNodes, err := splitStores(rev.GetNodes())
+	if err != nil {
+		return "", err
+	}
+	target := ""
+	if len(syncNodes) > 0 {
+		target = syncNodes[0]
```

Comment:
this logic will be different when we go to multiple rkv. story for another day.

Reply:
It is reading from the sync(local) nodes have fewest time latencies. The sync nodes are sorted by latencies.